### DISCUSSION TITLE
Product customization

### DISCUSSION THREAD

**Discussion by IT-Nomad** [PrestaShop/PrestaShop#36897] [Ideas]
### The problem

If a module uses `product customization` and sets in the  table 'PREFIX_customization_field' is_module=1 , every time the product is saved (after changing name,price etc ) using Prestashops Save and Publish, the product is saved with is_module=0 

### The solution

When editing a product, in Details TAB, for Customization, beside Name, type (text/file), checkbox (Required/Not required) should be also the checkbox IsModule (true/false). Missing this checkbox , always resets to `is_module=false`
or prestashop should ignore the customizations set with 'PREFIX_customization_field' is_module=1, this way would not mess up with modules.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Codencode** [comment]
I have also encountered this issue.
The problem occurs here:

https://github.com/PrestaShop/PrestaShop/blob/c6cf71fb6e8363b0b25fcbc450e25f7ae4cb0ced/src/Core/Form/IdentifiableObject/CommandBuilder/Product/CustomizationFieldsCommandsBuilder.php#L72-L79

where `added_by_module` is always set to false, so the value is overwritten.

"I will try to create a PR to fix the issue.

**Comment by Codencode** [comment]
This is the PR I have proposed: #36914
